### DISCUSSION TITLE
Including subfolders in folder downloads?

### DISCUSSION THREAD

**Discussion by julian45** [slskd/slskd#1051] [Ideas]
While browsing users' files, I've occasionally come across folders of music that also happen to contain subfolders (for example, scans from a CD booklet). Currently, I can only attempt to download the files in the top-level directory or just the subdirectory, not anywhere in between. Thus, if I'm (for example) downloading multiple albums that have Scans subfolders, all of those separate albums' files land in the same download folder, which can make things confusing.

I would greatly appreciate it if it were possible to include subfolders in download requests. Ideally, the download initiation would catch all levels of subdirectories (for example, folder A contains subfolders B1, B2, B3... each of which contain subfolders C1, C2, C3, and so on) along with any individual files in the selected folder, but if that's not possible for whatever reason, I'd still appreciate any degree of this that would be possible.

**Comment by jpdillingham** [comment]
Thanks for sharing!

The current "browse" UI is lacking, primarily because of performance issues, but it could also use some usability upgrades such as the one you're requesting.

Do you have an idea what this might look like?

**Reply by julian45** [reply]
I'm not sure how it might work in the slskd UI, but in the original Soulseek client, if I was browsing a user's files and happened to hit "Download" on a directory that had subdirectories, it added both that directory and all its subdirectories to the download queue, and preserved the folder hierarchy within the topmost directory (i.e., the one I originally selected). This worked well enough for me in that client, so that could possibly be one way of going about it.

**Reply by julian45** [reply]
I just came back to this repo for the first time in a while, and it looks like #1113 covers exactly the functionality I had in mind!
